### PR TITLE
Made external_crates feature work again.

### LINKED ISSUE #21928: liballoc doesn't compile with `--features external_crate` anymore
The error is:

    src/lib.rs:76:12: 76:16 error: unused or unknown feature, #[deny(unused_features)] on by default
    src/lib.rs:76 #![feature(libc)]

### PR DESCRIPTION
**Pull request by scialex**
Also added test for it.

Fixes #21928

### CODE REVIEW

File: src/liballoc/lib.rs [hunk 1]
```diff
@@ -78,7 +78,6 @@
 #[macro_use]
 extern crate core;
 
-#[cfg(all(not(feature = "external_funcs"), not(feature = "external_crate")))]
```
Comment (phil-opp) [inline src/liballoc/lib.rs]: That removes the error but links to libc instead. Adding `#[allow(unused_features)]` is a way better 'fix' (it still allows it globally...) without introducing a pointless platform specific dependency. IMHO the real problem is that it isn't possible to add `#![feature(libc)]` only when no other features are present...

Comment (scialex) [inline src/liballoc/lib.rs]: aggreed

**Comment by scialex** [comment]
@alexcrichton Could I get just a regular r+, This is a bug that actually breaks some code...

**Comment by alexcrichton** [comment]
@bors: rollup-

Sure thing!

**Pull request merged**